lib/ansible/config/manager.py

```diff
@@ -287,15 +287,16 @@ class ConfigManager(object):
 
     def __init__(self, conf_file=None, defs_file=None):
 
+        self._defs_file = defs_file or '%s/base.yml' % os.path.dirname(__file__)
```

Comment:
I think the other fix we discussed (fixing gather_facts not to mutate the config objects) is better than exposing these details to the unit tests.